firmware/src/display_task.cpp

```diff
@@ -35,9 +35,15 @@ void DisplayTask::run() {
     tft_.setRotation(SK_DISPLAY_ROTATION);
     tft_.fillScreen(TFT_DARKGREEN);
 
+  #ifdef CONFIG_IDF_TARGET_ESP32S3
+    ledcSetup(LEDC_CHANNEL_LCD_BACKLIGHT, 5000, 8);
```

Comment:
Hmmmm, I think this can be as high as 13 bit resolution on the S3? https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/ledc.html#ledc-api-supported-range-frequency-duty-resolution

Might be nice to pull the bit depth out as a constant so the code is the same with just some constants changed,  e.g.
```
    ledcSetup(LEDC_CHANNEL_LCD_BACKLIGHT, 5000, SK_BACKLIGHT_BIT_DEPTH);
    ledcAttachPin(PIN_LCD_BACKLIGHT, LEDC_CHANNEL_LCD_BACKLIGHT);
    ledcWrite(LEDC_CHANNEL_LCD_BACKLIGHT, (1 << SK_BACKLIGHT_BIT_DEPTH) - 1);
```
and set SK_BACKLIGHT_BIT_DEPTH to 16 and 13 for the different envs in platformio.ini (or even just set it at the top of this file based on CONFIG_IDF_TARGET_ESP32S3).



I think `DisplayTask::setBrightness` probably also needs to be updated as well to handle the reduced bit depth. I'm thinking keeping the brightness API as 16-bit (to keep the hardware differences encapsulated) and then map the 16-bit values to the supported bit depth, e.g.
```
brightness_ = brightness >> (16 - SK_BACKLIGHT_BIT_DEPTH);
```


Any chance you could test this change, since I don't have an S3-based SmartKnob View (yet)?

Reply:
Yes, I will test it and update PR. That is a good catch for the rest of the brightness variables 👍

Reply:
Updated PR, can you validate it with yours PCB? Just to be sure that it works as expected with PWM bits.